engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -98,7 +103,14 @@ class NextgenEngine @JvmOverloads constructor(
         }
 
         val executionContext = NadelExecutionContext(executionInput, serviceExecutionHooks)
-        val instrumentationContext = getInstrumentationContext(query, queryDocument, executionInput, overallSchema, instrumentation, instrumentationState)
+        val instrumentationContext = getInstrumentationContext(
```

Comment:
A little unrelated to this PR, but this function should really be refactored as it's not clear at all what it's doing i.e. it's calling `beginExecute` but would you really know that from the name `getInstrumentationContext` or from the args?

We should change `getInstrumentationContext` to an extension function instead

```kotlin
internal suspend fun NadelInstrumentation.beginExecute(
    query: ExecutableNormalizedOperation,
    queryDocument: Document,
    executionInput: ExecutionInput,
    graphQLSchema: GraphQLSchema,
    instrumentationState: InstrumentationState?,
): InstrumentationContext<ExecutionResult> {
    val nadelInstrumentationExecuteOperationParameters = buildInstrumentationExecutionParameters(
        query,
        queryDocument,
        executionInput,
        graphQLSchema,
        instrumentationState,
    )

    return beginExecute(nadelInstrumentationExecuteOperationParameters)
        .asDeferred()
        .await()
}
```

That way we're just calling `instrumentation.beginExecute(query, queryDocument, executionInput, graphQLSchema, instrumentationState)`

Reply:
very nice!